cpu/sam0_common/periph/spi.c

```diff
@@ -134,6 +134,16 @@ void spi_init_pins(spi_t bus)
     gpio_init_mux(spi_config[bus].miso_pin, spi_config[bus].miso_mux);
     gpio_init_mux(spi_config[bus].mosi_pin, spi_config[bus].mosi_mux);
     /* clk_pin will be muxed during acquire / release */
+
+    mutex_unlock(&locks[bus]);
```

Comment:
What is the logic for the mutex here?

Reply:
So if another thread tries to access the bus, it will be blocked until SPI is configured again.

Reply:
I understand the part of wanting to avoid `spi` to acquire the de-initiated pins, my question goes more in asking about a context where this is useful.

Reply:
Its more out of curiosity

Reply:
You are right that SPI devices are usually only accessed from one thread, but all other SPI functions make use of that mutex to ensure thread safety, so this just seemed like a logical extension.